lib/internal/test_runner/test.js

```diff
@@ -288,6 +288,13 @@ class Test extends AsyncResource {
     };
     this.waitingOn = 0;
     this.finished = false;
+
+    if (!testOnlyFlag && (only || this.runOnlySubtests)) {
```

Comment:
Please move this up with the rest of the validation. It could even be an `else` on [this conditional](https://github.com/nodejs/node/blob/e9bf7c1259f2ee24c1b238d665f9c6c681af917d/lib/internal/test_runner/test.js#L249-L251).

Reply:
I've moved it up as you've suggested. The reason it was here originally was due to the coupling with `this.diagnostic` & `this.diagnostics` which I've changed. Hopefully this is a bit clearer now.

Reply:
I've reverted back to the previous approach with the updated warning message https://github.com/nodejs/node/commit/56692925d354e6f24f0671f348baf44ae3c10585.